src/figuretype/trader.c

```diff
@@ -808,3 +854,78 @@ int figure_trade_sea_trade_units()
     }
     return 12;
 }
+
+void figure_trader_ship_record_dock(figure *ship, int dock_id) {
+    building *dock = building_get(dock_id);
+    dock->data.dock.trade_ship_id = ship->id;
+    for (int i = 0; i < 10; i++) {
+        if (dock_id == city_buildings_get_working_dock(i)) {
+            ship->building_id |= 1 << i;
+            return;
+        }
+    }
+}
+
+int figure_trader_ship_docked_once_at_dock(figure *ship, int dock_id) {
+    for (int i = 0; i < 10; i++) {
+        if (dock_id == city_buildings_get_working_dock(i)) {
+            if (ship->building_id & 1 << i) {
+                return 1;
+            }
+        }
+    }
+    return 0;
+}
+
+int figure_trader_ship_can_queue_for_import(figure *ship) {
+    if (ship->action_state != FIGURE_ACTION_112_TRADE_SHIP_MOORED) return 1;
```

Comment:
Again, avoid single-liners for ifs, use `||` instead

Reply:
fixed, FYI there's still few other single-liners ifs around `grep -rsn "if\b.*;" src/`